addons/purchase/tests/test_purchase.py

```diff
@@ -483,6 +483,46 @@ def test_purchase_multicurrency(self):
 
         self.assertEqual(order_b.order_line.price_unit, 10.0, 'The price unit should be 10.0')
 
+    def test_discount_update_when_no_seller_found(self):
+        """ Purchase order line should reset discount when no vendor found based on quantity
+        """
+        vendor = self.env['res.partner'].create({
+            'name': 'Test Vendor',
+            'supplier_rank': 1,
+        })
```

Comment:
If we can avoid record creation, it's better
Here you have `self.partner_a` and `self.partner_b`